.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "imported/llvm-dialects"]
```

Comment:
Should we be using a submodule when the rest of the GPUOpen driver is done with repo?

Reply:
I think so, yes. We unfortunately ended up in a situation where some of our workflows use repo and others (internally) use Git submodules.

It seems pretty clear to me that we cannot introduce repo to our internal workflows. Conversely, introducing Git submodules to the gpuopen workflows does not seem too problematic. Repo can fetch submodules fine, see https://github.com/GPUOpen-Drivers/AMDVLK/pull/297

So the options that are realistically on the table are:

* Use Git submodules everywhere
* Use repo for gpuopen and Git submodules internally

The advantages of Git submodules everywhere that I can see are:

* Git submodules make it easier to avoid version conflicts between LLPC and llvm-dialects because compatibility is tested explicitly as part of LLPC PRs that bump the llvm-dialects version. (This should be done manually as part of LLPC PRs that require new llvm-dialects features.)
* The same mechanism for llvm-dialects is used both internally and externally. Among other consequences, this reduces friction for having development, PRs, etc. on gpuopen.
* The bulk of LLPC contributions is by AMD employees who will all be familiar with Git submodules (while only a subset is familiar with repo).

The advantage of repo on gpuopen that I can see is that it may be less surprising for people who only know the gpuopen workflow.